framer.go

```diff
@@ -113,6 +149,14 @@ func (f *framer) AddActiveStream(id protocol.StreamID, str sendStreamI) {
 	f.mutex.Unlock()
 }
 
+func (f *framer) AddStreamWithControlFrames(id protocol.StreamID, str streamControlFrameGetter) {
+	f.controlFrameMutex.Lock()
+	if _, ok := f.streamsWithControlFrames[id]; !ok {
+		f.streamsWithControlFrames[id] = str
+	}
```

Comment:
Should we log if the stream is present in the map already?

Reply:
This is not indicative of any error. For example, the receive side of a stream could send a MAX_STREAM_DATA frame, while the send side of the wishes to send a RESET_STREAM frame.

There are also situations where the same side of the stream might enqueue itself multiple times: For example, the send side might want to send a BLOCKED frame, and the user calls `CancelWrite` right after that (which would make the stream register to send a RESET_STREAM frame).